src/driver/DriverUtils.ts

```diff
@@ -155,21 +170,28 @@ export class DriverUtils {
             password = usernameAndPassword.substr(firstColon + 1);
         }
 
+        // If replicaSet have value set It as hostlist, If not set like standalone host
         if (replicaSet) {
             hostReplicaSet = hostAndPort;
         } else {
             [host, port] = hostAndPort.split(":");
         }
 
-        return {
+        let connectionUrl: any = {
             type: type,
             host: host,
             hostReplicaSet: hostReplicaSet,
             username: decodeURIComponent(username),
             password: decodeURIComponent(password),
             port: port ? parseInt(port) : undefined,
-            database: afterBase || undefined,
-            replicaSet: replicaSet || undefined
+            database: afterBase || undefined
         };
+
+        // Loop to set every options in connectionUrl to object
+        for (const [key, value] of Object.entries(optionsObject)) {
+            connectionUrl[key] = value;
+        }
```

Comment:
While this is written well, its intent can be made even clearer by using a simple spread operator with `optionsObject` like so:

```ts
let connectionUrl: any = {
    type: type,
    host: host,
    hostReplicaSet: hostReplicaSet,
    username: decodeURIComponent(username),
    password: decodeURIComponent(password),
    port: port ? parseInt(port) : undefined,
    database: afterBase || undefined,
    ...optionsObject
};
```